### PR TITLE
Feat/nevermined xdai support

### PR DESCRIPTION
**Pull request by 0xArdi**
This PR adds support for purchasing nevermined subscriptions with XDAI.

### CODE REVIEW

File: packages/valory/skills/decision_maker_abci/behaviours/base.py [hunk 1]
```diff
@@ -561,6 +569,11 @@ def finish_behaviour(self, payload: BaseTxPayload) -> Generator:
 class BaseSubscriptionBehaviour(DecisionMakerBaseBehaviour, ABC):
     """Base class for subscription behaviours."""
 
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize `RedeemBehaviour`."""
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/behaviours/base.py:573]:
```suggestion
        """Initialize `BaseSubscriptionBehaviour`."""
```

File: packages/valory/skills/decision_maker_abci/behaviours/base.py [hunk 2]
```diff
@@ -620,6 +633,11 @@ def payment_token(self) -> str:
         subscription_params = self.subscription_params
         return subscription_params["payment_token"]
 
+    @property
+    def is_xdai(self) -> bool:
+        """Check if the payment token is xDAI"""
+        return self.payment_token == ZERO_ADDRESS
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/behaviours/base.py:639]: Can you explain this?

File: packages/valory/skills/decision_maker_abci/behaviours/base.py [hunk 3]
```diff
@@ -650,3 +668,34 @@ def _resolve_did(self) -> Generator[None, None, Optional[Dict[str, Any]]]:
...
+            contract_public_id=TransferNftCondition.contract_id,
+            contract_callable="balance_of",
+            data_key="data",
+            placeholder="balance",
+            address=address,
+            did=did,
+        )
+        if not result:
+            return False
+        return True
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/behaviours/base.py:688]:
```suggestion
        return result
```

File: packages/valory/skills/decision_maker_abci/behaviours/base.py [hunk 4]
```diff
@@ -650,3 +668,34 @@ def _resolve_did(self) -> Generator[None, None, Optional[Dict[str, Any]]]:
...
+        result = yield from self._get_nft_balance(
+            self.token_address,
+            self.synchronized_data.safe_contract_address,
+            zero_x_transformer(no_did_prefixed(self.did)),
+        )
+        if not result:
+            self.context.logger.warning("Failed to get balance")
+            return False
+
+        return self.balance > 0
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/behaviours/base.py:699]: Is your intention to loop here indefinitely until the balance is >0?

File: packages/valory/skills/decision_maker_abci/behaviours/base.py (hunk 2 above)
Comment (0xArdi) [reply packages/valory/skills/decision_maker_abci/behaviours/base.py:639]: When the payment token for the sbuscription is xdai (the native token of the chain), nevermined sets the payment address to the zeroAddress.

Comment (Adamantios) [reply packages/valory/skills/decision_maker_abci/behaviours/base.py:639]: Could you leave this as a comment in the code?

**Review by Adamantios — APPROVED** [review]

**Pull request merged**